src/api/handlers/deviceprofile.py

```diff
@@ -0,0 +1,139 @@
+"""Handler file for all routes pertaining to users"""
+from functools import wraps
+from _main_.utils.route_handler import RouteHandler
+from _main_.utils.massenergize_response import MassenergizeResponse
+from _main_.utils.context import Context
+from api.decorators import admins_only, super_admins_only, login_required
+from api.services.deviceprofile import DeviceService
+
+class DeviceHandler(RouteHandler):
+
+  def __init__(self):
+    super().__init__()
+    self.service = DeviceService()
+    self.registerRoutes()
+  
+  def __get_client_meta(self, request, args):
+    meta_data = request.META
+
+    x_forwarded_for = meta_data.get('HTTP_X_FORWARDED_FOR')
+    if x_forwarded_for:
+      args["ip_address"] = x_forwarded_for.split(',')[-1]
+    else:
+      args["ip_address"] = request.META.get('REMOTE_ADDR')
+    
+    # args["lang"] = meta_data.get('LANG') # TODO: we can choose to store language
+    # Example: { 'LANG': 'en_US.UTF-8' }
+    # args["time_zone"] = meta_data.get('TZ') # TODO: we can choose to store language
+    # Example: { 'TZ': 'UTC' }
+    # args["user_agent"] = meta_data.get('HTTP_USER_AGENT') # TODO: we can choose to store language
+    # args["device_type"] = meta_data.get('HTTP_USER_AGENT') # TODO: extract device type
```

Comment:
Why not extract and store the device type?

Reply:
I took that out for now because I found the formatting very confusing and I figured we could add it in later.

Reply:
I'm ok with just saving IP addresses, which was my idea before.  May be good enough for our purposes.  When I started looking at this before, I had looked at a package geoIP2 which had a database of locations from IP addresses.  This information could be useful since it provides a rough latitude/longitude for filtering by distance.  Just a thought, could be nice to add data to the DeviceProfile to plot where it is.